### PR TITLE
Cwarren5701/type system lesson

### PR DESCRIPTION
**Pull request by cwarren5701**
This is our first draft. We were unable to make a couple of the code blocks without the class/main method headers but we would like to not have those be visible. Also, in our bool section we kept and short and just made a button that should eventually link to ryan's bool lesson but right now it just links to the intro page.

### CODE REVIEW

**Comment by cloudthepain** [comment]
Looks Good

**Comment by cwarren5701** [comment]
Not sure why there are still 5 files changed because we had branches for those other things, including the variables lesson but for some reason it still say that those are changed files.

File: IntroToCSharp/Pages/Activities/TheTypeSystem.razor [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+<MudText Typo="Typo.h4" Style="font-weight:bold">The Type System</MudText>
+
+<Section Title="Introduction">
+    <MudText>As we learned in the <MudLink Href="/activities/variables">Variables</MudLink> lesson, each variables must
+        have a <Keyword Word="type"></Keyword> that describes its <Keyword Word="data"></Keyword>. There are <Keyword
+            Word="14"></Keyword> different types of data that programmers can use. Defining data types for our variables
+        allows the computer to know how to store it while our program runs.
+    </MudText>
+</Section>
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/TheTypeSystem.razor:12]: What do you think about:

```
In the Variables section we learned that each variable must have a type that describes the data that is being stored in the variable.
```

**Review by jcollard — COMMENTED** [review]
What do you think about creating a lesson called "Basic Data Types" that focuses on just `boolean`, `int`, `double`, and `string` (these are the most common "primitive" data types).

Then, the math lesson could go into "int", "double", and "decimal" in depth.

Another lesson specifically on "boolean" (Ryan has done a great job there).

Another lesson specifically on strings and characters.

Then finally an "optional" lesson "They Type System" which is more of a technical overview of all of the data types we didn't talk about and how they're not as common.

The Basic Data Types lesson would preceed this lesson "The Type System" lesson which would allow you to be a lot more technical on this page and make it an "optional" lesson.

File: IntroToCSharp/Pages/Activities/TheTypeSystem.razor [hunk 2]
```diff
@@ -0,0 +1,151 @@
+@inherits CaptainCoder.Page
+@page "/activities/the-type-system"
+
+<MudText Typo="Typo.h4" Style="font-weight:bold">The Type System</MudText>
+
+<Section Title="Introduction">
+    <MudText>As we learned in the <MudLink Href="/activities/variables">Variables</MudLink> lesson, each variables must
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/TheTypeSystem.razor:7]: You should favor `<Keyword Word="type"/>` over `<Keyword Word="type"></Keyword>`

File: IntroToCSharp/Pages/Activities/TheTypeSystem.razor [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    </MudElement>
+    <MudText>Float and double values are not exactly precise, they are simply estimations of those defined values.
+        But a decimal value is precise. You will see this in the below example.</MudText>
+
+    <CodeBlock ReplIt="JoshuaJoshJosh1/FloatPointExample">
+    using System;
+class FloatingPointNumbers
+{
+	public static void Main()
+	{
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/TheTypeSystem.razor:54]: Let's stick to using examples without a `Main` method:

```csharp
		 float floatNumber = 1.5f;
 		 double doubleNumber = 1.5;
   		 decimal decimalNumber = 1.5m;
  		 Console.WriteLine(floatNumber - 1.0f);
   		 Console.WriteLine(doubleNumber - 1);
   		 Console.WriteLine(decimalNumber - 1.0m);
```

File: IntroToCSharp/Pages/Activities/TheTypeSystem.razor [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    </Questions>
+</Section>
+
+<Section Title="Strings">
+    <MudText>A string is text of any length. Generally speaking, strings contain letters, punctation, and a couple other
+        symbols. Strings are often used in programs for names, descriptions, or large bodies of text.</MudText>
+
+    <MudText>A string is made by placing text inside double quotes like this:</MudText>
+    <CodeBlock>
+        string playerName = "Jimmy";
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/TheTypeSystem.razor:85]: Remove indentation for consistency.

File: IntroToCSharp/Pages/Activities/TheTypeSystem.razor (hunk 2 above)
Comment (josht77) [reply IntroToCSharp/Pages/Activities/TheTypeSystem.razor:7]: done

File: IntroToCSharp/Pages/Activities/TheTypeSystem.razor (hunk 3 above)
Comment (josht77) [reply IntroToCSharp/Pages/Activities/TheTypeSystem.razor:54]: All of the answers come out to 0.5, so need to fix this example or change it completely

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+<Section Title="Characters">
+    <MudText>
+        In C#, the <Keyword Word="char"/> type represents a single character. This
+        is most often used when creating text using a <Keyword Word="string"/>. The
+        char data type is a special type of integer for which each integer value
+        represents a symbol to be displayed. The char data type uses 2 bytes of
+        data which means it has 65,536 distinct characters.
+    </MudText>
+    <Note>The characters that can be displayed with the </Note>
```
Comment (cloudthepain) [inline IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:42]: Is there something that is supposed to be here?

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        different number of <Keyword Word="byte" />s as well as being <Keyword Word="signed"/> or
+        <Keyword Word="unsigned"/>.
+    </MudText>
+    <Note>You do not need to memorize this table but it is good to be aware.</Note>
+    <DataTable Headers="@intHeaders" Rows="@intRows"></DataTable>
+</Section>
+
+<Section Title="Characters">
+    <MudText>
+        In C#, the <Keyword Word="char"/> type represents a single character. This
```
Comment (cloudthepain) [inline IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:36]: This is pretty wordy and gets a little confusing reading it.

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+<Section Title="Floating-Point Numbers">
+    <MudText>
+        Integers are great for whole numbers, but what happens when we want to
+        store decimals or even repeating decimals like π? Floating-point numbers
+        are a data type that allows programmers to <b>approximate</b> a real number.
+    </MudText>
+    <MudElement Class="ma-4">
+        <DataTable Headers=@floatHeaders Rows=@floatRows/>
+    </MudElement>
+    <MudText>
```
Comment (cloudthepain) [inline IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:84]: I would add in a line that mentions how the Literal indicators are used.

Comment (cloudthepain) [reply IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:84]: You also could consider adding how to constrain to a certain amount of decimals places.

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+Console.WriteLine($"Decimal: {dec}");
+    </CodeBlock>
+    <MudText>
+        In the above example, you might expect the result of all three values to
+        be <Code>0.1</Code>. However, because float and double are
+        approximations, we see a <Keyword Word="rounding error"/> in the output.
+    </MudText>
+</Section>
+
+<Section Title="Booleans">
```
Comment (cloudthepain) [inline IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:107]: Not sure if you want to add that booleans are either 0 or 1 as well.

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor (hunk 5 above)
Comment (jcollard) [reply IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:42]: Nope! Thanks.

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor (hunk 8 above)
Comment (jcollard) [reply IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:107]: In C# this is not the case.

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor (hunk 6 above)
Comment (jcollard) [reply IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:36]: How about this?

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor (hunk 7 above)
Comment (jcollard) [reply IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:84]: I'm going to leave the constraints for another section on formatting text.

Comment (jcollard) [reply IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:84]: What do you think about the additions I made to this sextion?

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+        different number of <Keyword Word="byte" />s as well as being 
+        <Keyword Word="signed"/> or <Keyword Word="unsigned"/>.
+    </MudText>
+    <Note>You do not need to memorize this table but it is good to be aware.</Note>
+    <DataTable Headers="@intHeaders" Rows="@intRows"></DataTable>
+</Section>
+
+<Section Title="Characters">
+    <MudText>
+        To represent text, we use the <Keyword Word="string"/> data type. This
```
Comment (cloudthepain) [inline IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:30]: This reads a lot more clearly now.

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor (hunk 7 above)
Comment (cloudthepain) [reply IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:76]: I think that this is a good addition. The flow of this page feels a lot more fluid and clear now.

File: IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+        declared using the <Keyword Word="char"/> data type. The
+        <Code>char</Code>data type is stored as 2 bytes of data resulting in
+        65,536 distinct characters.
+    </MudText>
+    <MudText>
+        Unlike a <Code>string</Code>, a <Code>char</Code> is created by placing
+        a single character inside of single quotes like this:
+    </MudText>
+    <CodeBlock AllowCopy>char someLetter = 'j';</CodeBlock>
+    <Note>You must place <b>exactly</b> on character between the single quotes.</Note>
```
[the remark this answers is not in the document]
Comment (jcollard) [reply IntroToCSharp/Pages/Lessons/Variables/TheTypeSystem.razor:41]: :+1:

**Pull request merged**